examples_apps/run_minimizing_correlations.py

```diff
@@ -0,0 +1,355 @@
+# the initial state of mode ``q[1]``, which we will optimize over. Note
+# that we take the outer product
+# :math:`\ket{in}=\ket{\psi}\otimes\ket{\phi}`, and use the ``Ket``
+# operator to initialise the circuit in this initial multimode pure state.
+
+
+def cost(phi_var_re, phi_var_im):
+    phi_var = tf.cast(phi_var_re, tf.complex64) + 1j*tf.cast(phi_var_im, tf.complex64)
+
+    in_state = tf.einsum('i,j->ij', psi, phi_var)
+
+    result = eng.run(prog, args={"input_state": in_state}, modes=[1])
+    state = result.state
+
+    rhoB = state.dm()
+    cost = tf.cast(tf.math.real((tf.linalg.trace(rhoB @ rhoB)
+                                -penalty_strength*tf.linalg.trace(rhoB @ x)**2
+                                -penalty_strength*tf.linalg.trace(rhoB @ p)**2)
+                           /(tf.linalg.trace(rhoB))**2), tf.float64)
+
+    return cost
+
+
+######################################################################
+# The cost function runs the engine, and we use the argument ``modes=[1]`` to
+# return the density matrix of mode ``q[1]``.
+#
+# The returned cost contains the purity of the reduced density matrix
+#
+# .. math:: \text{Tr}(\rho_B^2),
+#
+# and an extra penalty that forces the optimized state to have zero
+# displacement; that is, we want to minimise the value
+#
+# .. math:: \langle \hat{x}\rangle=\text{Tr}(\rho_B\hat{x}).
+#
+# Finally, we divide by the :math:`\text{Tr}(\rho_B)^2` so that the state
+# is always normalized.
+#
+# We can now set up the optimization, to minimise the cost function.
+# Running the optimization process for 200 reps:
+#
+
+opt = tf.keras.optimizers.Adam(learning_rate=0.007)
+
+reps = 200
+
+cost_progress = []
+
+for i in range(reps):
+    # reset the engine if it has already been executed
+    if eng.run_progs:
+        eng.reset()
+
+    with tf.GradientTape() as tape:
+        loss = -cost(phi_var_re, phi_var_im)
+
+    # Stores cost at each step
+    cost_progress.append(loss.numpy())
+
+    # one repetition of the optimization
+    gradients = tape.gradient(loss, [phi_var_re, phi_var_im])
+    opt.apply_gradients(zip(gradients, [phi_var_re, phi_var_im]))
+
+    # Prints progress
+    if i % 1 == 0:
+        print("Rep: {} Cost: {}".format(i, loss))
+
+
+######################################################################
+# We can see that the optimization converges to the optimum purity value
+# of 0.9122365.
+#
+# Visualising the optimum state
+# -----------------------------
+#
+# We can now calculate the density matrix of the input state
+# :math:`\ket{\phi}` which minimises entanglement:
+#
+# .. math:: \rho_{\phi} = \ket{\phi}\left\langle \phi\right|
+#
+
+ket_val = phi_var_re.numpy() + phi_var_im.numpy()*1j
+out_rhoB = np.outer(ket_val, ket_val.conj())
+
+
+######################################################################
+# Next, we can use the following function to plot the Wigner function of
+# this density matrix:
+#
+
+import copy
```

Comment:
I guess now that is a python script we can no longer just import qutip here, right?

Reply:
Oh, the original jupyter notebook wasn't importing Qutip either? I didn't make any changes to this section

Reply:
It did, but I was hoping it could be removed.